Examples/intermediate_examples/simulation_intermediate_demo.py

```
@@ -42,19 +49,31 @@
     print("Please ensure ManipulaPy is properly installed and accessible.")
     print("Current Python path:", sys.path)
     sys.exit(1)
 
+# PyBullet is optional: the physics-simulation section degrades gracefully to a
+# skip when it (or a usable physics server) is unavailable.
+try:
+    import pybullet as pb
+    from ManipulaPy.sim import Simulation
+
+    PYBULLET_AVAILABLE = True
+except Exception:  # pragma: no cover - depends on optional dependency
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)